src/window.cpp

```diff
@@ -1818,6 +1831,8 @@ void UnInitWindowSystem()
 {
 	UnshowCriticalError();
 
+	_closed_windows.clear();
+
```

Comment:
Might it be useful to have some logic that `assert`s when the `Window` is `delete`d from an unsafe location? I guess that would need to be in `~Window`. That way any remaining `delete`s get caught, also in further development of other windows as people might not have been accustomed to this new "API". It would also help preventing/catching similar problems in patch packs.